### PR TITLE
Refactor staff window to class

### PR DESCRIPTION
**Pull request by IntelOrca**

### CODE REVIEW

File: src/openrct2-ui/windows/StaffList.cpp [hunk 1]
```diff
@@ -112,648 +78,625 @@ static rct_widget window_staff_list_widgets[] = {
...
-        case WIDX_STAFF_LIST_UNIFORM_COLOUR_PICKER:
-            WindowDropdownShowColour(
-                w, widget, w->colours[1], staff_get_colour(static_cast<StaffType>(_windowStaffListSelectedTab)));
-            break;
-    }
-}
+            // Enable highlighting of these staff members in map window
+            if (window_find_by_class(WC_MAP) != nullptr)
+            {
+                gWindowMapFlashingFlags |= (1 << 2);
```
Comment (duncanspumpkin) [inline src/openrct2-ui/windows/StaffList.cpp:190]: When you rebase please use the new flags for this.

File: src/openrct2-ui/windows/StaffList.cpp [hunk 2]
```diff
@@ -112,648 +78,625 @@ static rct_widget window_staff_list_widgets[] = {
...
-    auto i = 0;
-    for (auto spriteIndex : StaffList)
+    void DrawTabImages(rct_drawpixelinfo& dpi)
     {
-        if (y > dpi->y + dpi->height)
+        // Handymen tab image
+        auto selectedTab = _selectedTab;
+        auto i = (selectedTab == 0 ? (_tabAnimationIndex & ~3) : 0);
+        i += GetPeepAnimation(PeepSpriteType::Handyman).base_image + 1;
+        i |= SPRITE_ID_PALETTE_COLOUR_1(gStaffHandymanColour);
```
Comment (duncanspumpkin) [inline src/openrct2-ui/windows/StaffList.cpp:513]: can you use a better name than `i`. This isn't an iterator.

File: src/openrct2/peep/Staff.cpp [hunk 3]
```diff
@@ -2677,3 +2665,19 @@ void Staff::UpdateRideInspected(ride_id_t rideIndex)
             | RIDE_INVALIDATE_RIDE_LIST;
     }
 }
+
+money32 GetStaffWage(StaffType type)
```
Comment (duncanspumpkin) [inline src/openrct2/peep/Staff.cpp:2669]: Do you prefer this over a table? Maybe we should make a StaffTypeDescriptor table same as rides and tracks.

Comment (IntelOrca) [reply src/openrct2/peep/Staff.cpp:2669]: Generally yes because it doesn't rely on enum order

File: src/openrct2-ui/windows/StaffList.cpp (hunk 2 above)
Comment (Gymnasiast) [reply src/openrct2-ui/windows/StaffList.cpp:513]: I think we use `imageId` in most places.

File: src/openrct2-ui/windows/StaffList.cpp [hunk 4]
```diff
@@ -112,648 +78,625 @@ static rct_widget window_staff_list_widgets[] = {
...
-    gfx_draw_sprite(
-        dpi, i,
-        w->windowPos
-            + ScreenCoordsXY{ (window_staff_list_widgets[WIDX_STAFF_LIST_SECURITY_TAB].left
-                               + window_staff_list_widgets[WIDX_STAFF_LIST_SECURITY_TAB].right)
-                                  / 2,
-                              window_staff_list_widgets[WIDX_STAFF_LIST_SECURITY_TAB].bottom - 6 },
-        0);
-
-    rct_drawpixelinfo sprite_dpi;
```
Comment (duncanspumpkin) [inline src/openrct2-ui/windows/StaffList.cpp:603]: `entertainerTabDpi`

File: src/openrct2-ui/windows/StaffList.cpp [hunk 5]
```diff
@@ -112,648 +78,625 @@ static rct_widget window_staff_list_widgets[] = {
...
-    auto y = 0;
-    auto i = 0;
-    for (auto spriteIndex : StaffList)
+    void DrawTabImages(rct_drawpixelinfo& dpi)
     {
-        if (y > dpi->y + dpi->height)
+        // Handymen tab image
+        auto selectedTab = _selectedTab;
+        auto imageId = (selectedTab == 0 ? (_tabAnimationIndex & ~3) : 0);
+        imageId += GetPeepAnimation(PeepSpriteType::Handyman).base_image + 1;
```
Comment (duncanspumpkin) [inline src/openrct2-ui/windows/StaffList.cpp:512]: I feel the tab drawing should be simplified into a loop but maybe for some later time.

**Review by duncanspumpkin — APPROVED** [review]

**Pull request merged**